### ISSUE TITLE
firebase.messaging().requestPermission() not showing Permission request popup on Android

### ISSUE THREAD

**Issue by theoryweary** [invertase/react-native-firebase-starter#80]
I just installed the react-native-firebase-starter package and tried to request Notification permissions with this:

const messagingRequestPermission = () => {
  console.log('firebase messaging requesting permission')
  return firebase.messaging().requestPermission()
    .then(() => {
       console.log('requestPermission - user has permission')
      return true
    })
    .catch(error => {
            console.log('requestPermission -  does not have permission')
      return false
    });
}

In the console I see this output, showing that RequestPermission returned true
   firebase messaging requesting permission 
   requestPermission - user has permission

However, on Android, I never see a popup when starting the app asking me to give the app permission to send notifications.   On iOS I enabled Push Notifications in XCode and I do receive the generic iOS notifications message. I can also schedule notifications on iOS and Android but I do not actually receive them (I tested this on real devices with the app closed when the notification should have been shown).

I'm copying my react-native code from another project where notifications are working using react-native-firebase which I manually installed.  I had other build issues with that project though so I thought I'd rebuild everything on top of the react-native-firebase-starter. As far as I can tell my AndroidManifest.xml files are the same.   Seems like this could be an issue with the starter build?

Any help greatly appreciated!!

**Comment by theoryweary** [comment]
Hi, just an update.   According to Firebase my app has permission to use messaging, but it is still not asking for permission using the dialogue box that i saw when I did the manual react-native-firebase setup.  Any ideas how this makes sense?

Here's what I'm using to check that firebase has messaging permission: 
  firebase.messaging().hasPermission()
  .then(enabled => {
    if (enabled) {
      console.log('firebase says user has notification permission')
      // user has permissions
    } else {
      console.log('firebase says user does not have notification permission')
      // user doesn't have permission
    }
  });

**Comment by krystofbe** [comment]
I have these permissions
```
    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" /> 
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
    <uses-permission android:name="android.permission.VIBRATE" />
```

`~/Library/Android/sdk/tools/emulator -avd Nexus_5X_API_26`
Android 8.0.0
![image](https://user-images.githubusercontent.com/12120657/49518853-f349f900-f89f-11e8-82c2-6224fa9ed356.png)

**Comment by krystofbe** [comment]
I see that registration with firebase worked and i got a FCM token for my android device. it just doesnt process/show any android notifications.

**Comment by theoryweary** [comment]
I have no idea what I did, but notifications started working for me on Android a week after I started this thread.  Sorry that isn't more helpful, here are the permissions I have in my manifest:

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.READ_CONTACTS"/>
    <uses-permission android:name="com.android.vending.BILLING" />

**Comment by Shivendra30** [comment]
Even I have the same issue
The FCM Token is generated but there is no popup that asks the user for the permission

**Comment by onkar-dange** [comment]
I have also same issue
there is no popup for permission

**Comment by bradgreens** [comment]
To add something hopefully of value, it's worth noting push notification permissions on Android are enabled by default. So any conditional logic will follow the happy path _of not_ prompting the user.

Then, it's worth noting if you're in the emulator the permissions prompt seems pretty fragile (that is, I haven't triggered it _yet_) and push notifications appear difficult to configure with the emulator (I'm also trying to get that working, but am resorting to a physical USB attached device now)

**Comment by arapocket** [comment]
Have the same issue. But i'll try implementing the notifications if you are right @bradgreens

**Comment by phcbarros** [comment]
I have the same issue =/, it works on iOs but not on Android. I'm testing in a real device (Moto Z Play 2 Android 8.0.0). Does anyone has some idea how to solve this?

I have this configuration
macOS Mojave v: 10.14.2
react-native-cli: 2.0.1
react-native: 0.57.8
node: 10.13.0
yarn: 1.12.3
react-native-firebase: 5.2.3

build.gradle
```gradle
buildscript {
    ext {
        buildToolsVersion = "28.0.2"
        minSdkVersion = 16
        compileSdkVersion = 27
        targetSdkVersion = 26
        supportLibVersion = "27.1.1"
    }
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.2.0'
        classpath 'com.google.gms:google-services:4.0.1'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        google()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
    }
}


task wrapper(type: Wrapper) {
    gradleVersion = '4.6'
    distributionUrl = distributionUrl.replace("bin", "all")
}
```

app/build.gradle
```gradle
dependencies {
    implementation 'me.leolin:ShortcutBadger:1.1.21@aar'
    implementation "com.google.firebase:firebase-analytics:16.0.6"
    implementation 'com.android.support:support-v4:27.1.1'
    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
    implementation 'com.google.firebase:firebase-config:16.1.2'
    implementation "com.google.firebase:firebase-firestore:17.1.0" 
    implementation "com.google.firebase:firebase-messaging:17.3.1"
}
```
AndroidManifest.xml
```xml
<uses-permission android:name="android.permission.INTERNET" />
<uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW" />
<uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
<uses-permission android:name="android.permission.VIBRATE" />
<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
<uses-permission android:name="android.permission.CAMERA" />
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
```

**Comment by SMKH-PRO** [comment]
RNfirebase.messaging().hasPermission()
  .then(enabled => {
              if (enabled) {
                // user has permissions
               console.log("have");
               this.SavePushToken()
    } else {
      // user doesn't have permission
      console.log("dont have");
      RNfirebase.messaging().requestPermission()
      .then((res) => {
        // User has authorised 
        this.SavePushToken()  

      })
      .catch(error => {
        // User has rejected permissions  
this.NoNotifPermision() //alert no notic permission

      });
    } 
  });

Now the problem here is that, Request permission will never ask for permission and returns .then no matter the permissions are really granted or not or even blocked.  it is always returning then, it should reject instead

**Comment by costerik** [comment]
Here the same issue

**Comment by ngoctan95** [comment]
And so now, it has not been resolved yet? 
Incredible bug. :|

**Comment by a-c-sreedhar-reddy** [comment]
Did anyone solve this?

**Comment by ngoctan95** [comment]
Just resolved this bug by using another library for checking permission Android only and show modal for opening system setting screen.

**Comment by a-c-sreedhar-reddy** [comment]
@ngoctan95  which library

**Comment by Akila1928** [comment]
Any updates on this? I have no way to show the consent screen in android.

**Comment by shyaminayesh** [comment]
> by using another library for checking per

can you explain little bit ? which library you used to resolve this ?

**Comment by pearlismylove** [comment]
Same here~

**Comment by ChalomEllezam** [comment]
Here a good post on this.

https://stackoverflow.com/questions/44305206/ask-permission-for-push-notification

**Comment by filipef101** [comment]
TLDR: You don't have to ask for Push Notification permissions on Android. 

@theoryweary You can close the issue

**Comment by theoryweary** [comment]
@filipef101  is right - notifications are actually not considered a dangerous permission on Android so there is no need to ask the user for permission.

**Comment by bbeckk** [comment]
If the notification permission is disabled manuallyby user in android, we need to ask for notification permission, shouldn't we? Since requestPermission always gives true in android, how can we do that?

**Comment by fkotsian** [comment]
@bbeckk Haven't implemented this myself yet, but as far as I can tell the permissions to "allow receiving notifications" and "show notifications" on Android are different. 

The device permissions to "receive notifications" are provided by default, but the user can turn off "show me notifications". 

You'd have to redirect the user to the Android Settings screen to turn "show me notifications" back on, at least as I currently understand it. Sorry it's not super helpful but will update once I implement
